apps/namadillo/src/App/Masp/ShieldedFungibleTable.tsx

```diff
@@ -1,84 +1,121 @@
-import { ActionButton, TableRow } from "@namada/components";
-import { formatPercentage } from "@namada/utils";
-import { AtomErrorBoundary } from "App/Common/AtomErrorBoundary";
-import { NamCurrency } from "App/Common/NamCurrency";
+import { Asset } from "@chain-registry/types";
+import { Balance } from "@heliaxdev/namada-sdk/web";
+import { ActionButton, Currency, TableRow } from "@namada/components";
+import { formatCurrency, formatPercentage } from "@namada/utils";
 import { TableWithPaginator } from "App/Common/TableWithPaginator";
 import { routes } from "App/routes";
+import { chainTokensAtom } from "atoms/chain";
+import { knownChainsAtom } from "atoms/integrations/atoms";
 import BigNumber from "bignumber.js";
-import { AtomWithQueryResult } from "jotai-tanstack-query";
-import { useEffect, useState } from "react";
+import { useAtomValue } from "jotai";
+import { useEffect, useMemo, useState } from "react";
 import { twMerge } from "tailwind-merge";
+import namadaShieldedSvg from "./assets/namada-shielded.svg";
 
-type MockedToken = {
+type TokenRow = {
   name: string;
-  balance: number;
-  dollar: number;
-  ssrRate: number;
+  icon?: string;
+  balance: BigNumber;
+  dollar: BigNumber;
+  ssrRate: BigNumber;
 };
 
-const mockData = (): AtomWithQueryResult<MockedToken[]> =>
-  ({
-    data: [
-      {
-        name: "NAM",
-        balance: 9999.99,
-        dollar: 9999.99,
-        ssrRate: 0.99,
-      },
-      {
-        name: "ATOM",
-        balance: 9999.99,
-        dollar: 9999.99,
-        ssrRate: 0.99,
-      },
-    ],
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  }) as any;
-
 const resultsPerPage = 100;
 const initialPage = 0;
 
-export const ShieldedFungibleTable = (): JSX.Element => {
+const namadaAsset: Asset = {
```

Comment:
Can we move this, temporarily, to a new file, just like we have for registry/namada.json ? I think it will be super useful in other parts of the codebase